OpenRA.Mods.Common/Traits/BotModules/SquadManagerBotModule.cs

```diff
@@ -128,7 +128,7 @@ public bool IsEnemyUnit(Actor a)
 
 		public bool IsNotHiddenUnit(Actor a)
 		{
-			return a.TraitsImplementing<Cloak>().Where(ha => ha.IsVisible(a, Player)).Any();
+			return !a.TraitsImplementing<IVisibilityModifier>().Any() || a.TraitsImplementing<IVisibilityModifier>().Where(ha => ha.IsVisible(a, Player)).Any();
```

Comment:
Performance suggestion: LINQ is expensive and trait lookups are, too, so how about something like
```
var hasModifier = false;
var visModifiers = a.TraitsImplementing<IVisibilityModifier>().ToArray();
foreach (var v in visModifiers)
{
	if (v.IsVisible(a, Player))
		return true;

	hasModifier = true;
}

return !hasModifier;
```

Reply:
I don't think you actually have to/should `ToArray` in this case since we're only enumerating once.

Reply:
OK dude, valuable suggetions.